docs/sources/operator/architecture.md

```diff
@@ -43,6 +43,24 @@ A single resource can belong to multiple hierarchies. For example, if two
 GrafanaAgents use the same Probe, modifying that Probe will cause both
 GrafanaAgents to be reconciled.
 
+### Build the Hierarchy
+
+The hierarchy is built via label matching on the custom resources. Below is a
+graph illustrating the matching. The `GrafanaAgent` picks up the `MetricsInstance`
+and `LogsInstance` that match the labels `app.kubernetes.io/name: loki` and `app.kubernetes.io/instance: release-name`. The instances pick up the resources the same way.
```

Comment:
We probably should use different labels for an example.

Reply:
What would be good labels for the example?  Might want to do it now because it might slip through the cracks...

Reply:
I believe the pattern is very common. Unfortunately I don't know the space well enough.

Reply:
@captncraig Do you know what labels we could use instead?

Reply:
In the quickstart we use:

```
  metrics:
    instanceSelector:
      matchLabels:
        agent: grafana-agent-metrics
  logs:
    instanceSelector:
      matchLabels:
        agent: grafana-agent-logs
```

and 

```
podMonitorSelector:
    matchLabels:
      instance: primary
```

I'd usually recommend using the `name` of the parent object as the label value. And it looks like we have been using `instance` as the key. Maybe we should standardize on a more namespaced label key, but thats a different discussion entirely.